### PR TITLE
ONNX support, bug fix, user friendly APIs

### PR DESCRIPTION
**Pull request by hhsecond**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,5 +20,6 @@
 $ pip install redisai
 ```
 
+[RedisAI example repo](https://github.com/RedisAI/redisai-examples) shows few examples made using redisai-py under python client section.
```
Comment (lantiga) [inline README.md]: Minor nit: 
```
The [RedisAI examples repo](https://github.com/RedisAI/redisai-examples) includes a few examples made using redisai-py in the `python_client` directory.
```

Comment (hhsecond) [reply README.md]: Done

**Comment by hhsecond** [comment]
@lantiga Added few more things to this PR
- Saving both onnx models and sklearn models are possible now. If sklearn, we convert using skl2onnx
- `save_model`/`load_model` APIs are available in the top level itself.
- More tests

**Comment by lantiga** [comment]
@hhsecond ONNXRuntime support has been merged, we should now fix the failing test here (I tried rerunning figuring that it had to do with the fact that the ONNXRuntime PR in RedisAI had not been merged yet, but that's not the case)

**Comment by lantiga** [comment]
Ok, the error is just `unsupported backend`
```
test_run_onnxdl_model (test.ClientTestCase) ... ERROR

======================================================================
ERROR: test_run_onnxdl_model (test.ClientTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/circleci/repo/test/test.py", line 113, in test_run_onnxdl_model
    con.modelset("onnx_model", Backend.onnx, Device.cpu, onnxdl_model)
  File "/home/circleci/repo/redisai/client.py", line 192, in modelset
    return self.execute_command(*args)
  File "/home/circleci/repo/venv/lib/python3.7/site-packages/redis/client.py", line 775, in execute_command
    return self.parse_response(connection, command_name, **options)
  File "/home/circleci/repo/venv/lib/python3.7/site-packages/redis/client.py", line 789, in parse_response
    response = connection.read_response()
  File "/home/circleci/repo/venv/lib/python3.7/site-packages/redis/connection.py", line 642, in read_response
    raise response
redis.exceptions.ResponseError: unsupported backend
```

So apparently the tests are not running against the latest docker container (`redisai:edge`)

**Comment by lantiga** [comment]
@gkorland For situations like this, it would be useful if we could test against nightly, rather than latest. What do you think?

**Comment by gkorland** [comment]
@lantiga 
We have them both, we test on commit with latest and run nightly with edge, check the circleci config file

**Comment by lantiga** [comment]
@gkorland Right, but this means that we need wait for a feature to go into release for tests to pass, as in this case. Wouldn't it be a better option to run tests on edge?

**Comment by gkorland** [comment]
Edge is built from master, Latest is built from release

**Comment by lantiga** [comment]
Correct. What I'm suggesting is to configure the project so that tests for `redisai-py` run against master (i.e. `edge`) redisai instead of `latest`.

**Comment by gkorland** [comment]
But that's what are doing we're testing redisai-py with latest and edge

**Comment by hhsecond** [comment]
We test on `edge` as a nightly job (scheduled). On each commit, we are running only on `latest` and using that as a criterion for the merge. Since commits to this repository have a dependency on the nightly of `redisai` (usually), It would make sense to reconfigure the build so that each commit could also run on `edge` instead of `latest`.

**Comment by lantiga** [comment]
> Based on my understanding of the circleci config, we test on edge as a nightly job (scheduled). But on each commit, we are running only on latest and using that as a criterion for the merge. Since commits to this repository have a dependency on the master of redisai (usually), It would make sense to reconfigure the build so that each commit could run on edge instead of latest. What do you think?

@gkorland that was my thinking, sorry if I wasn't clear. I would reconfigure the build so that tests run on edge, so we can make changes in tandem between redisai and redisai-py without waiting for a release.

**Comment by lantiga** [comment]
@hhsecond @gkorland I'd like to proceed making `redisai:edge` what we test against, so we can merge the PR

**Comment by gkorland** [comment]
@lantiga I guess till we release 1.0.0 we should stay with the edge

**Comment by hhsecond** [comment]
@gkorland I just pushed the updated config file but edge in docker hub is two days old. Is that why it's breaking?

**Comment by hhsecond** [comment]
We had some issues with the module imports in the CI. Fixed it and all the tests are passing. @lantiga Let's merge this?

**Comment by lantiga** [comment]
Awesome, thank you @hhsecond

**Pull request merged**